llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -50049,8 +50049,8 @@ static SDValue combineAndNotOrIntoAndNotAnd(SDNode *N, SelectionDAG &DAG) {
   SDValue X, Y, Z;
   if (sd_match(N, m_And(m_Value(X),
                         m_OneUse(m_Or(m_Value(Y), m_Not(m_Value(Z))))))) {
-    // Don't fold if Y is a constant to prevent infinite loops.
-    if (!isa<ConstantSDNode>(Y))
```

Comment:
Should probably be `isConstantIntBuildVectorOrConstantInt` for both `Y` and `N`.

Reply:
Do you mean `Y` and `Z`?

Reply:
Switched both `isa<ConstantSDNode>` casts to `isConstantIntBuildVectorOrConstantInt` 👍